src/nvim/CMakeLists.txt

```
@@ -160,6 +160,8 @@ elseif(CMAKE_SYSTEM_NAME MATCHES "Darwin")
   # ENABLE_EXPORTS is set to true. See
   # https://github.com/neovim/neovim/issues/25295
   target_link_options(nvim_bin PRIVATE "-Wl,-export_dynamic")
+  # Apple Clang 16's new deduplication pass leads to lock-up on start (as of beta 2)
```

Just so we know when we can possible remove this?

```suggestion
  # Apple Clang 16's new deduplication pass leads to lock-up on start (as of beta 2)
  # TODO(clason): test if this is still needed once Apple Clang 16 is not on beta.
```